eng/Update-Snippets.ps1

```diff
@@ -0,0 +1,11 @@
+$generatorProject = "$PSScriptRoot\SnippetGenerator\SnippetGenerator.csproj";
+dotnet build $generatorProject
+
+foreach ($file in Get-ChildItem "$PSScriptRoot\..\sdk" -Filter README.md -Recurse)
```

Comment:
Have we considered potentially keeping snippets in their own files, so that each can be a small and contained test.    I'm not a fan of `regions`, personally, and worry about polluting the sample code with them will make things less clear to the casual reader.

Reply:
Would you want to inject the entire file with headers and everything?

Reply:
I was thinking more of keeping the region-based approach, but segmenting it from the samples proper.  In some form that can be run and validated as part of the nightly process, but maybe a set of tests in its own folder or something similar.

Reply:
I'm open to improvements but wanted to start with something simple

Reply:
I'm not sure that mine is the best suggestion either.   One of the goals that we had discussed was helping developers understand the copy/paste worthy areas of the samples, which the regions do - if (and only if) you understand the convention.

Too bad that we can't create our own alias for `region` and use something like:

```csharp

var code = ...

#snippet Do the thing

// Stuff

#endsnippet

Reply:
I know that @danieljurek is planning to start digging into samples and help define a convention so that they can essentially be standalone examples and can be ran during our smoke testing.  Personally also avoid using regions but instead just use the entire method body. We can easily attribute the method with the name for the sample so that we can align them with the readmes.

Reply:
Also by attribute we don't actually have to depend on a .NET Attribute type but we can simply add a structured comment on the method and look for that as well.

Reply:
I think using the full method would definitely favor not using the full samples, lest our ``ReadMe` files become a novel.    It could work in something isolated with a very intentional scoping to snippet use.

Reply:
DocFX uses regions that's why I went with them, it would be easy to reuse code when writing doc articles.